service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -360,4 +364,33 @@ protected void extractIfMatchVersionParam(final RoutingContext ctx) {
         }
         ctx.next();
     }
+
+    /**
+     * Creates default CORS handler that allows 'POST', 'GET', 'PUT' and 'DELETE' methods for the specified origin.
+     *
+     * @param allowedOrigin The allowed origin pattern.
+     * @return The handler.
+     */
+    protected final CorsHandler createDefaultCorsHandler(final String allowedOrigin) {
+        return createCorsHandler(allowedOrigin, EnumSet.of(
+                HttpMethod.POST,
+                HttpMethod.GET,
+                HttpMethod.PUT,
+                HttpMethod.DELETE)
+        );
+    }
+
+    /**
+     * Creates CORS Handler that allows HTTP methods for the specified origin.
+     *
+     * @param allowedOrigin The allowed origin pattern.
+     * @param methods Set of allowed HTTP methods
+     * @return The handler.
+     */
+    protected final CorsHandler createCorsHandler(final String allowedOrigin, final Set<HttpMethod> methods) {
+        return CorsHandler.create(allowedOrigin)
+                .allowedMethods(methods)
+                .allowedHeader(HttpHeaders.CONTENT_TYPE.toString())
+                .allowedHeader(HttpHeaders.AUTHORIZATION.toString());
```

Comment:
we will also need to expose the *ETag* header and allow the *If-Match* header ...